### ISSUE TITLE
Use Custom DNS

### ISSUE THREAD

**Issue by robin98** [Qv2ray/Qv2ray#1170]
Hello guys;
I tried to use custom DNS server while I connected to a proxy, but I failed! The software, tunnels the whole traffic, So it seems I can't set my own DNS server for DNS queries!
I tried from Preferences > DNS Settings and add my DNS there. also I tried from Groups > DNS Settings. but no one overwrite DNS Setting and use my own LOCAL DNS SERVER runs at 127.0.0.1:53 at the same computer I ran Qv2ray.

**Comment by DuckSoft** [comment]
![图片](https://user-images.githubusercontent.com/7822648/102693045-989b5400-4252-11eb-92ac-d9c7ddb37cb7.png)

**Comment by robin98** [comment]
It Unchecked by default! But I enable it and check for any changes. but nothing at all!

**Comment by DuckSoft** [comment]
Show us the evidence. Why do you think this "is not working"?
At least show us the log.

Also:

```
    This program is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
    GNU General Public License for more details.
```

**Comment by MarksonHon** [comment]
@robin98 Go to the V2Ray documentation
<https://www.v2fly.org/en_US>

**Comment by DuckSoft** [comment]
Read the fancy docs before blaming anyone: https://www.v2fly.org/config/dns.html#dnsobject
![图片](https://user-images.githubusercontent.com/7822648/102693606-bd91c600-4256-11eb-8f93-d2e5ee408bd2.png)
What should I say? It worked perfectly on my computer. Clear all your DNSes and write `127.0.0.1` there.
You should reflect on yourself rather on the software.

**Comment by DuckSoft** [comment]
@robin98 Check the comment above.

**Comment by robin98** [comment]
@DuckSoft Ok buddy 😑 don't kick me for that, I just ask because I don't familiar with change options in advanced. Here is my steps until now:

1. Running my desired DNS Service on the `127.0.0.1:53`.
2. I went to `Preferences > DNS Settings`. Clear all DNS entries and just add `127.0.0.1`.
3. Check `Connections Settings` tab. the `Use Local DNS` option was **Un**checked by default, so I do nothing here.
4. I opened the Firefox, go to `Network Settings` and add `127.0.0.1:1089` as the SOCKS host. Also I checked `Proxy DNS when using SOCKS v5` option.
5. going to dnsleaktest.com, and click on the `Standard test`.
6. The results will show me the DNS Address for Proxy server(same location and same IP as proxy server), not that one I hosted on `127.0.0.1`.
